### PR TITLE
Revamp wasm demo

### PR DESCRIPTION
**Pull request by edwin0cheng**
This PR try to use a newer version of RA, which implemented:

* Upgrade dependencies for rust and js.
* Use newer RA (which locked to https://github.com/rust-analyzer/rust-analyzer/commit/27befe6c7fe064b364182e2ad54825b5e5f9dee3)
* Because RA used `rayon`,  I tried to use [wasm-bindgen-rayon](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon) and it works ! (And move the whole RA state to Web Worker)
* There is no need to change any code RA side. (Thanks for our `denyc` CI flag)
* However, there is a caveat : In order to support web thread, some [http response headers](https://github.com/GoogleChromeLabs/wasm-bindgen-rayon#final-caveats) has to enabled.  (Chrome still works without the headers for now, but it would change this behavior in May)
* Due to this reason, it cannot be hosted in github pages which do no allow custom headers.
* Maybe we could setup a netlify account and host in there ? (Just like how [wasm-bindgen](https://rustwasm.github.io/wasm-bindgen/examples/raytrace.html) raytracer demo [do](https://wasm-bindgen.netlify.app/exbuild/raytrace-parallel/).)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -8,18 +8,17 @@ edition = "2018"
...
+serde-wasm-bindgen = "0.1.3"
+stacker = "0.1.13"
+wasm-bindgen = "0.2.72"
+wasm-bindgen-rayon = "1.0.1"
 
-ra_ide_api = { git = "https://github.com/rust-analyzer/rust-analyzer", features = ["wasm"] }
-ra_syntax =  { git = "https://github.com/rust-analyzer/rust-analyzer" }
-ra_text_edit =  { git = "https://github.com/rust-analyzer/rust-analyzer" }
+ide = { git = "https://github.com/rust-analyzer/rust-analyzer", rev = "27befe6c7" }
+ide_db = { git = "https://github.com/rust-analyzer/rust-analyzer", rev = "27befe6c7" }
```
Comment (matklad) [inline Cargo.toml]: Can/should we used auto-published crates here?

File: src/conv.rs [hunk 2]
```diff
@@ -86,7 +98,7 @@ impl Conv for Severity {
     }
 }
 
-impl ConvWith<&LineIndex> for &AtomTextEdit {
+impl ConvWith<&LineIndex> for &Indel {
```
Comment (matklad) [inline src/conv.rs]: Urgh, we probably should replace it with to_proto/from_proto pattren, like we did in rust-analyzer proper.

**Review by matklad — COMMENTED** [review]
I am torn on mutitherading. On the one hand, the easiest thing to do is to say that we don't use parallelism on wasm. On the other hand, rayon in the browser is just cool, and, if/when  we bundle stdlib, parallelism would matter. 

My gut feel is that we should try netlify or something similar and see how much hassle it creates. So, :+1: for me on creating a netlify account.

File: src/conv.rs (hunk 2 above)
Comment (edwin0cheng) [reply src/conv.rs]: Fixed.

File: Cargo.toml (hunk 1 above)
Comment (edwin0cheng) [reply Cargo.toml]: Sure, and it works 🎉

**Comment by edwin0cheng** [comment]
> My gut feel is that we should try netlify or something similar and see how much hassle it creates. So, +1 for me on creating a netlify account.

Yeah, and I would work on neflify CI in this PR too, so I convert this PR to draft first.

**Comment by edwin0cheng** [comment]
Okay, I have setup a netlify CI in this PR and which will auto deploy to https://ra-wasm.netlify.app/  on pushing to master branch.

One thing we need to setup here is Github Secrets, @matklad  I will pm you in Zulip for that.

**Comment by matklad** [comment]
@edwin0cheng granted you admin permissions for this repo!

**Comment by edwin0cheng** [comment]
Everything should work now, let merged .

**Pull request merged**